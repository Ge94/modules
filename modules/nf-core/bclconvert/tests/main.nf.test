nextflow_process {

    name "Test Process BCLCONVERT"
    script "../main.nf"
    process "BCLCONVERT"
    config "./nextflow.config"

    tag "bclconvert"
    tag "modules"
    tag "modules_nfcore"

    test("homo_sapiens illumina [bcl]") {
        when {
            process {
                //TODO use new test dataset when available, see https://github.com/nf-core/test-datasets/issues/996
                """
                input[0] = [
                    [ id: 'test', lane:1 ],
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/bcl/flowcell_samplesheet.csv', checkIfExists: true),
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/bcl/flowcell.tar.gz', checkIfExists: true)
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
                    process.out.fastq,
                    process.out.fastq_idx,
                    process.out.undetermined.collect { meta, fastq -> file(fastq).name },
                    process.out.undetermined_idx,
                    process.out.reports,
                    process.out.logs.collect { meta, logs -> file(logs).list().sort() },
                    process.out.interop.collect { meta, interop ->
                                                interop.findAll { interopfile ->
                                                file(interopfile).name != "IndexMetricsOut.bin" } },
                    process.out.versions
                ).match() },
                { assert process.out.interop.collect { meta, interop ->
                                                        interop.find { interopfile -> file(interopfile).name == "IndexMetricsOut.bin" } } }
            )
        }
    }

    test("homo_sapiens illumina [bcl] - stub") {
        options "-stub"

        when {
            process {
                //TODO use new test dataset when available, see https://github.com/nf-core/test-datasets/issues/996
                """
                input[0] = [
                    [ id: 'test', lane:1 ],
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/bcl/flowcell_samplesheet.csv', checkIfExists: true),
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/bcl/flowcell.tar.gz', checkIfExists: true)
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
}
