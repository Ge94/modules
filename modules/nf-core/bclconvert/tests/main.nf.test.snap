{
    "homo sapiens illumina [bcl] - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "lane": 1
                        },
                        "Sample1_S1_L001_R1_001.fastq.gz:md5,68b329da9893e34099c7d8ad5cb9c940"
                    ]
                ],
                "1": [
                    
                ],
                "2": [
                    [
                        {
                            "id": "test",
                            "lane": 1
                        },
                        "Undetermined_S0_L001_R1_001.fastq.gz:md5,68b329da9893e34099c7d8ad5cb9c940"
                    ]
                ],
                "3": [
                    
                ],
                "4": [
                    [
                        {
                            "id": "test",
                            "lane": 1
                        },
                        [
                            "Adapter_Cycle_Metrics.csv:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "Adapter_Metrics.csv:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "Demultiplex_Stats.csv:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "Demultiplex_Tile_Stats.csv:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "IndexMetricsOut.bin:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "Index_Hopping_Counts.csv:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "Quality_Metrics.csv:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "Quality_Tile_Metrics.csv:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "RunInfo.xml:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "SampleSheet.csv:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "Top_Unknown_Barcodes.csv:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "fastq_list.csv:md5,d41d8cd98f00b204e9800998ecf8427e"
                        ]
                    ]
                ],
                "5": [
                    [
                        {
                            "id": "test",
                            "lane": 1
                        },
                        [
                            "Errors.log:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "FastqComplete.log:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "Info.log:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "Warnings.log:md5,d41d8cd98f00b204e9800998ecf8427e"
                        ]
                    ]
                ],
                "6": [
                    [
                        {
                            "id": "test",
                            "lane": 1
                        },
                        [
                            "ControlMetricsOut.bin:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "CorrectedIntMetricsOut.bin:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "ErrorMetricsOut.bin:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "ExtractionMetricsOut.bin:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "IndexMetricsOut.bin:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "QMetricsOut.bin:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "TileMetricsOut.bin:md5,d41d8cd98f00b204e9800998ecf8427e"
                        ]
                    ]
                ],
                "7": [
                    "versions.yml:md5,3742a25082bc9f0ff8dca303c5c8cb8b"
                ],
                "fastq": [
                    [
                        {
                            "id": "test",
                            "lane": 1
                        },
                        "Sample1_S1_L001_R1_001.fastq.gz:md5,68b329da9893e34099c7d8ad5cb9c940"
                    ]
                ],
                "fastq_idx": [
                    
                ],
                "interop": [
                    [
                        {
                            "id": "test",
                            "lane": 1
                        },
                        [
                            "ControlMetricsOut.bin:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "CorrectedIntMetricsOut.bin:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "ErrorMetricsOut.bin:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "ExtractionMetricsOut.bin:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "IndexMetricsOut.bin:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "QMetricsOut.bin:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "TileMetricsOut.bin:md5,d41d8cd98f00b204e9800998ecf8427e"
                        ]
                    ]
                ],
                "logs": [
                    [
                        {
                            "id": "test",
                            "lane": 1
                        },
                        [
                            "Errors.log:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "FastqComplete.log:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "Info.log:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "Warnings.log:md5,d41d8cd98f00b204e9800998ecf8427e"
                        ]
                    ]
                ],
                "reports": [
                    [
                        {
                            "id": "test",
                            "lane": 1
                        },
                        [
                            "Adapter_Cycle_Metrics.csv:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "Adapter_Metrics.csv:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "Demultiplex_Stats.csv:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "Demultiplex_Tile_Stats.csv:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "IndexMetricsOut.bin:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "Index_Hopping_Counts.csv:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "Quality_Metrics.csv:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "Quality_Tile_Metrics.csv:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "RunInfo.xml:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "SampleSheet.csv:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "Top_Unknown_Barcodes.csv:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "fastq_list.csv:md5,d41d8cd98f00b204e9800998ecf8427e"
                        ]
                    ]
                ],
                "undetermined": [
                    [
                        {
                            "id": "test",
                            "lane": 1
                        },
                        "Undetermined_S0_L001_R1_001.fastq.gz:md5,68b329da9893e34099c7d8ad5cb9c940"
                    ]
                ],
                "undetermined_idx": [
                    
                ],
                "versions": [
                    "versions.yml:md5,3742a25082bc9f0ff8dca303c5c8cb8b"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.1"
        },
        "timestamp": "2024-11-20T06:23:25.978009664"
    },
    "homo sapiens illumina [bcl]": {
        "content": [
            [
                [
                    {
                        "id": "test",
                        "lane": 1
                    },
                    "Sample1_S1_L001_R1_001.fastq.gz:md5,0675fb6365322eaafb33c0f8e862b54b"
                ]
            ],
            [
                
            ],
            [
                [
                    {
                        "id": "test",
                        "lane": 1
                    },
                    "Undetermined_S0_L001_R1_001.fastq.gz:md5,d41d8cd98f00b204e9800998ecf8427e"
                ]
            ],
            [
                
            ],
            [
                [
                    {
                        "id": "test",
                        "lane": 1
                    },
                    [
                        "Adapter_Cycle_Metrics.csv:md5,5a0c88793b4a0885fe3dda16609b576e",
                        "Adapter_Metrics.csv:md5,989240b8840b2169ac1061f952c90f6c",
                        "Demultiplex_Stats.csv:md5,93949a8cd96f907d83e0808c1ec2a04b",
                        "Demultiplex_Tile_Stats.csv:md5,83120160b0f22a1303fa1db31c19f6e9",
                        "IndexMetricsOut.bin:md5,9e688c58a5487b8eaf69c9e1005ad0bf",
                        "Index_Hopping_Counts.csv:md5,1059369e375fd8f8423c0f6c934be978",
                        "Quality_Metrics.csv:md5,6614accb1bb414fe312b17b81f5521f7",
                        "Quality_Tile_Metrics.csv:md5,cdc89fd2962bdd4a24f71e186112118a",
                        "RunInfo.xml:md5,03038959f4dd181c86bc97ae71fe270a",
                        "SampleSheet.csv:md5,dc0dffd39541dd6cc5b4801d768a8d2b",
                        "Top_Unknown_Barcodes.csv:md5,2e2faba761137f228e56bd3428453ccc",
                        "fastq_list.csv:md5,ae1470cd741e33b87a4ad26f849a50f9"
                    ]
                ]
            ],
            [
                [
                    "Errors.log",
                    "FastqComplete.txt",
                    "Info.log",
                    "Warnings.log"
                ]
            ],
            [
                [
                    "ControlMetricsOut.bin:md5,6d77b38d0793a6e1ce1e85706e488953",
                    "CorrectedIntMetricsOut.bin:md5,2bbf84d3be72734addaa2fe794711434",
                    "ErrorMetricsOut.bin:md5,38c88def138e9bb832539911affdb286",
                    "ExtractionMetricsOut.bin:md5,7497c3178837eea8f09350b5cd252e99",
                    "QMetricsOut.bin:md5,7e9f198d53ebdfbb699a5f94cf1ed51c",
                    "TileMetricsOut.bin:md5,83891751ec1c91a425a524b476b6ca3c"
                ]
            ],
            [
                "versions.yml:md5,3742a25082bc9f0ff8dca303c5c8cb8b"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.1"
        },
        "timestamp": "2024-11-20T05:48:23.502941225"
    }
}